network/database.go

```diff
@@ -27,37 +26,86 @@ type StorageInterface struct {
 	storage.InjectBase
 }
 
+// Database prefixes:
+// Processes:       network:tree/<PID>
+// DNS Requests:    network:tree/<PID>/dns/<ID>
+// IP Connections:  network:tree/<PID>/ip/<ID>
+
+func makeKey(pid int, scope, id string) string {
+	if scope == "" {
+		return "network:tree/" + strconv.Itoa(pid)
+	}
+	return fmt.Sprintf("network:tree/%d/%s/%s", pid, scope, id)
+}
+
+func parseDBKey(key string) (pid int, scope, id string, ok bool) {
+	// Split into segments.
+	segments := strings.Split(key, "/")
+	// Check for valid prefix.
+	if !strings.HasPrefix("tree", segments[0]) {
+		return 0, "", "", false
+	}
+
+	// Keys have 2 or 4 segments.
+	switch len(segments) {
+	case 4:
+		id = segments[3]
+
+		fallthrough
+	case 3:
+		scope = segments[2]
+		// Sanity check.
+		switch scope {
+		case "dns", "ip", "":
+			// Parsed id matches possible values.
+			// The empty string is for matching a trailing slash for in query prefix.
+			// TODO: For queries, also prefixes of these values are valid.
+		default:
+			// Unknown scope.
+			return 0, "", "", false
+		}
+
+		fallthrough
+	case 2:
+		var err error
+		if segments[1] == "" {
+			pid = process.UndefinedProcessID
+		} else {
+			pid, err = strconv.Atoi(segments[1])
+			if err != nil {
+				return 0, "", "", false
+			}
```

Comment:
shouldn't we log that error? We expect to be able to parse it as a number so this is likely something that points to a bigger issue

Reply:
Yup. Checked.

Reply:
Ah, wrong comment. This will need some reworking anyway, we'll should have a chat about that.